### PR TITLE
fix: URI validation not executed for Replace OData prompting

### PR DESCRIPTION
**Pull request by mmilko01**
Fix for https://github.com/SAP/open-ux-tools/issues/2268.

- Adds URI validations for Replace OData prompting.

### CODE REVIEW

**Review by nikmace — APPROVED** [review]
Code looks good
Test coverage is excellent
Did not test manually

File: packages/adp-tooling/src/prompts/change-data-source/index.ts [hunk 1]
```diff
@@ -3,7 +3,45 @@ import type { ManifestNamespace } from '@sap-ux/project-access';
...
+    return true;
+}
+
+/**
+ * Validates the Annotation URI prompt.
+ *
+ * @param value The value to validate.
+ * @returns {boolean | string} True if the value is a valid URI or empty, or an error message if not a valid URI.
+ */
+function validatePromptAnnotationURI(value: string): boolean | string {
```
Comment (longieirl) [inline packages/adp-tooling/src/prompts/change-data-source/index.ts:33]: Would it make sense to try to join these two methods as the only difference is if `validationResult` is a string or a boolean?

Comment (mmilko01) [reply packages/adp-tooling/src/prompts/change-data-source/index.ts:33]: I see your point, but I didn't see a nice way to merge to the functions and not make the code uglier. I would either have to pass a parameter to know for which prompt the new functions is executing or have some logic inline in the prompt to parse the result. In both cases the I wouldn't have the nice looking passing of the validator functions directly in the prompts as it currently is. If you have an idea that I don't currently see, please suggest :)

Comment (longieirl) [reply packages/adp-tooling/src/prompts/change-data-source/index.ts:33]: Happy with explanation to cover the existing implementation. Thanks.

**Review by longieirl — APPROVED** [review]
Changes look good.
Comments addressed.
Tests updated.
Changeset present.

**Pull request merged**